docs/20_smart-contracts/40_guides/10_create_smart_contracts_guide.md

```diff
@@ -0,0 +1,1351 @@
+warning: transaction executed locally, but may not be confirmed by the network yet         ] 
+#         hello <= hello::getstats              ""
+>> Stats not initialized.
+```
+
+Call the `hi` action again:
+
+```shell
+dune --send-action hello hi '[ama]' hello@active
+```
+
+```txt
+executed transaction: cc41c0ebfe6449fcf5ecdacdba2edce0fe68551308a8250943b173f314659ae3  104 bytes  435 us
+warning: transaction executed locally, but may not be confirmed by the network yet         ] 
+#         hello <= hello::hi                    {"nm":"ama"}
+>> Name: ama, balance: 0.9990 EOS.
+#         hello <= hello::gettime               ""
+>> Year: 2023, Month: 2, Day: 26, Hour: 22, Minute: 3, Second: 34
+```
+
+And if you check the stats now:
+
+```cpp
+dune --send-action hello getstats '[]' hello@active
+```
+
+You can see the stats restarted from scratch:
+
+```txt
+executed transaction: 62e7853cf93654592badba9578f05182d10994d826b0f3bab093d47f83f73140  96 bytes  366 us
+warning: transaction executed locally, but may not be confirmed by the network yet         ] 
+#         hello <= hello::getstats              ""
+>> Stats for `hi` action: 1
+```
+
+## Indexes
+
+Indexes provide efficient and flexible access to data stored in multi-index tables. An index is a specialized data structure that allows you to look up data in the table based on a certain field, or a combination of fields. Indexes can be used to optimize the performance of queries that retrieve data from the table, and can also enforce uniqueness constraints on the data (only the primary index).
+
+EOS supports two types of indexes:
+
+- primary indexes and
+- secondary indexes.
+
+### Primary Indexes
+
+A primary index in EOS is a unique identifier for each row in the multi-index table. It is defined explicitly using the `primary_key()` member function. This function must be defined within the `struct` that represents the table, and must return a value that uniquely identifies each row. In the `hello` smart contract we already have a primary index defined for the `user_balance` table structure definition.
+
+```cpp
+uint64_t primary_key() const { return user.value; }
+```
+
+### Secondary Indexes
+
+A secondary index is any additional field in the table's structure that can be used to efficiently search and filter the data.
+Secondary indexes may be defined on data members which are not unique as well as unique ones. There can be up to 16 secondary indexes. Secondary indices support the following types:
+
+- `uint64_t`
+- `uint128_t`
+- `uint256_t`
+- `double`
+- `long double`
+
+If you add a secondary index to an existing multi-index table it will have unpredictable outcome.
+
+**Advance topic**: Consult the [Data design and migration](https://docs.eosnetwork.com/cdt/latest/best-practices/data-design-and-migration) documentation for more details on how to extended existing, already deployed, multi-index tables.
```

Comment:
This will need to have new guides written too, @VEG-Ginna please create a ticket for this so we can capture it.

Reply:
https://github.com/eosnetworkfoundation/devrel/issues/202